src/FSharp.Core/prim-types.fs

```diff
@@ -4069,6 +4072,28 @@ namespace Microsoft.FSharp.Core
 
     and 'T voption = ValueOption<'T>
 
+#if !NET8_0_OR_GREATER
```

Comment:
Lose the
'#if !NET8_0_OR_GREATER',

FSharp.Core is only built NETSTANDARD20 and NETSTANDARD21, so the attribute will be there.  The C# compiler just cares that an attribute exists, not that it exists in System.Runtime.  So long as the attribute remains internal, we should be fine.  The problems we had with the IsReadOnlyAttribute was due to a different framework library shipping the attribute as a public type.

Reply:
Yeah, the only reason to keep the ifdef would be as a reminder to remove it if we ever target something higher someday.

There is already this in the same file, for example:

https://github.com/dotnet/fsharp/blob/43bcb29ca5dd0958db7636133d92b60eb61cd9a6/src/FSharp.Core/prim-types.fs#L384-L385

Reply:
I removed it in 94a8541.